viz_scripts/plots.py

```diff
@@ -176,6 +176,20 @@ def energy_impact(x,y,color,plot_title,file_name):
     plt.legend(labels=objects, handles=patches, loc='upper right', borderaxespad=0, fontsize=15, frameon=True)
     plt.savefig(SAVE_DIR+file_name+".png", bbox_inches='tight')
 
+def barplot_sensed_mode(data,x,y,plot_title, labels, file_name):
+    colours = dict(zip(labels, plt.cm.tab20.colors[:len(labels)]))
```

Comment:
Why do we need this function? This is identical to `barplot_mode`. Why not just use `barplot_mode`?

Reply:
Yes, I did realize the function is identical to `barplot_mode`. But I decided to keep an exclusively different function for the `sensed` chart generation, despite being identical because:

1. Following the existing pattern of function definition: The other existing function definition for `pie_chart_mode` is also identical to the `pie_chart_sensed_mode` function. This led me to think, if we need to make changes exclusively impacting the `sensed` charts, and not the `non-sensed` charts; having two distinct functions would be good decision.

This function `barplot_sensed_mode` can be modifiable for changes related with `sensed` charts, without having any impact for the `non-sensed` charts generated in other notebooks. 

I understand having two identical functions `barplot_sensed_mode` & `barplot_mode` brings up code-duplication.  But I see a trade-off for the approach of having two functions will help us leverage easy modification in future.

Please let me know what would be the right approach in this case.
I think, if there is a probability of having different ways in which `sensed` chart and `non-sensed` charts would be generated, it seems like a good trade-off to keep two identical functions. But if these two charts generation would always be identical, I think using a single function is a good idea.

Reply:
Based on our discussion, I feel it's ideal to remove code-duplication and not engineer for future. 
I have removed the redundant `barplot_sensed_mode()`, and replaced the call with `barplot_mode()`.

Reply:
Documenting the discussion here for the future.

> But I see a trade-off for the approach of having two functions will help us leverage easy modification in future.
> But if these two charts generation would always be identical, I think using a single function is a good idea.

I do not see how this will leverage easy modification in the future. If we do need to modify in the future, we will still need to test the modified implementation, whether we generate the new implementation by copying the existing implementation at that time or now. Once we know what the potential changes are, we may also choose to implement the changes by refactoring the current implementation instead of copying, it depends on the changes required. This is pretty much the textbook definition of DRY and over-engineering, and should be removed now.